### DISCUSSION TITLE
v4 status

### DISCUSSION THREAD

**Discussion by tangjinzhou** [vueComponent/ant-design-vue#6212] [Show and tell]
# v4.0

## Overview:
   
refactor by cssinjs，follow the latest design specifications of antd 5.x

如果你想参与，请直接留言认领，但不承诺一定的合并， review存在主观意见，不针对任何人，不要玻璃心。
单次单人建议只认领一个组件，如多个组件相互依赖可同时认领。
认领后，请及时 pr 到 feat-v4 分支，如长时间没有进度或动态，会视为放弃。建议及时关注项目动态，避免重复工作。
同步内容可参考 antd 5.1.6 vs 4.21.5

可参考已完成组件的 cssinjs 使用方式。

## Progress

- [x] Affix @tangjinzhou 
- [x] Anchor @tangjinzhou 
- [x] Alert @tangjinzhou 
- [x] Avatar @tangjinzhou 
- [x] BackTop @tangjinzhou 
- [x] Badge @tangjinzhou 
- [x] Breadcrumb @tangjinzhou 
- [x] Button @tangjinzhou 
- [x] Col @zkwolf 
- [x] Divider @buqiyuan 
- [x] Icon @buqiyuan 
- [x] Layout @buqiyuan 
- [x] Menu @tangjinzhou 
- [x] Space @buqiyuan 
- [x] Typography @shifeng1993 
- [x] Row @zkwolf 
- [x] Spin @aibayanyu20 
- [x] Skeleton @aibayanyu20 
- [x] Comment @shifeng1993 
- [x] ConfigProvider @tangjinzhou 
- [x] Empty @buqiyuan 
- [x] Result @buqiyuan 
- [x] Descriptions @zkwolf 
- [x] PageHeader @shifeng1993 
- [x] DatePicker @shifeng1993 
- [x] Transfer @shifeng1993 
- [x] Switch @aibayanyu20 
- [x] Tooltip @tangjinzhou 
- [x] Mentions @aibayanyu20 
- [x] Progress @aibayanyu20 
- [x] Rate @songsong0707 
- [x] Statistic @shifeng1993 
- [x] List @shifeng1993 
- [x] Tag @buqiyuan 
- [x] Slider @shifeng1993 
- [x] Pagination @shifeng1993 
- [x] Popconfirm @tangjinzhou 
- [x] Popover @tangjinzhou 
- [x] Input @shifeng1993 
- [x] Card @webvs2 
- [x] Checkbox @shifeng1993 
- [x] Dropdown @tangjinzhou 
- [x] message @tangjinzhou 
- [x] notification @tangjinzhou 
- [x] Modal @tangjinzhou 
- [x] Upload @shifeng1993 
- [x] Carousel @shifeng1993 
- [x] Timeline @shifeng1993 
- [x] Steps @shifeng1993 
- [x] Collapse @shifeng1993 
- [x] InputNumber @shifeng1993 
- [x] Table @shifeng1993 
- [x] Tabs @webvs2 
- [x] Calendar @shifeng1993 
- [x] Tree @shifeng1993 
- [x] TimePicker @shifeng1993 
- [x] TreeSelect @shifeng1993 
- [x] Select @shifeng1993 
- [x] Cascader @tangjinzhou 
- [x] AutoComplete @tangjinzhou 
- [x] Image @tangjinzhou 
- [x] Drawer @tangjinzhou 
- [x] Radio @shifeng1993 
- [x] Form @tangjinzhou 

// new components
- [x]  segmented @selicens
- [x]  WaterMark @aibayanyu20 
- [x]  QrCode @selicens
- [x]  FloatButton @shifeng1993 
- [x] Tour @ykcory

**Comment by wsxe9988** [comment]
Tour（漫游式引导）组件
design tokens of Ant Design v5 to customize component
QrCode组件
WaterMark 水印组件
FloatButton 组件
segmented 分段控制器组件

**Reply by tangjinzhou** [reply]
单次单人建议只认领一个组件，如多个组件相互依赖可同时认领。

**Reply by tangjinzhou** [reply]
https://github.com/vueComponent/ant-design-vue/pull/6173 
segmented 已有一个 pr，会合并这个pr后迭代

**Reply by wsxe9988** [reply]
暂时不认领，只是把和ant design v5对标后缺失的组件列出来

**Comment by zkwolf** [comment]
- Descriptions #6213
- Row Col #6220

**Comment by buqiyuan** [comment]
- Divider  #6214
 - Icon  #6215
 - Layout #6249
 - Space #6229
 - Empty #6230
 - Result #6246
 - Tag #6227

**Reply by tangjinzhou** [reply]
Space 可以优先搞一下，这个好多组件有依赖

**Reply by buqiyuan** [reply]
好

**Comment by aibayanyu20** [comment]
* Spin #6222 
* Skeleton #6224 
* Progress #6234
* Switch #6236
* Mentions #6255
* WaterMark #6300

**Comment by shifeng1993** [comment]
- Typography #6244 
- Comment #6238 
- PageHeader #6239 
- Calendar #6269 
- DatePicker #6245 
- Transfer #6247 
- Statistic #6240 
- List #6241 
- TimePicker #6270 
- Tree #6276 
- Slider #6250 
- Pagination #6251 
- Input  #6237 
- Checkbox #6248 
- Upload #6261 
- Carousel #6262 
- TreeSelect #6296 
- Timeline #6263 
- Steps #6264 
- Collapse #6266 
- Table #6267 
- InputNumber #6265
- Select #6295 
- Radio #6299 

// new components
- FloatButton #6294 
- Tour #6332

**Reply by shifeng1993** [reply]
白天在公司是内网，搞不了，只能晚上加班完了回去整，有点慢见谅啊

**Reply by tangjinzhou** [reply]
很棒了

**Comment by songsong0707** [comment]
- Rate #6254

**Reply by tangjinzhou** [reply]
有进展吗，如时间不允许，可以及时释放认领

**Reply by shifeng1993** [reply]
我这会闲着，要不我一弄？

**Reply by songsong0707** [reply]
我前两天整了整，发现按钮组的样式不大对劲，还在改，让别的靓仔先弄吧

**Comment by webvs2** [comment]
- Card  #6258
- Tabs  #6288

**Comment by selicens** [comment]
- segmented https://github.com/vueComponent/ant-design-vue/pull/6286  

- qrcode https://github.com/vueComponent/ant-design-vue/pull/6315
- app https://github.com/vueComponent/ant-design-vue/pull/6719

**Reply by tangjinzhou** [reply]
@selicens 新组件认领直接编辑这个就好了 ，不要回复邮件

**Comment by wsxe9988** [comment]
ant design 5.5  终于出了ColorPicker这个新组件了

**Comment by CCherry07** [comment]
- ColorPicker https://github.com/vueComponent/ant-design-vue/pull/6722
